src/source/CppGenerator.scala

```diff
@@ -127,10 +150,20 @@ class CppGenerator(spec: Spec) extends Generator(spec) {
 
     val skipFirst = SkipFirst()
     for (c <- consts) {
-      skipFirst{ w.wl }
-      w.w(s"${marshal.fieldType(c.ty)} const $selfName::${idCpp.const(c.ident)} = ")
-      writeCppConst(w, c.ty, c.value)
-      w.wl(";")
+      val isConstExpr = c.value match {
```

Comment:
It would be nice to unify the cases for this match with the ones above in the header file.  One option would be to make a helper function which returns the constValue used in the header file, then have isConstExpr here simply test whether that value is non-empty.